gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -233,114 +413,57 @@ def run_translation_averaging(
+            ta_metrics = compute_metrics(set(w_i2Ui1_dict_inliers.keys()), i2Ui1_dict, wRi_list, wti_list, gt_wTi_list)
         else:
             ta_metrics = None
 
         num_translations = sum([1 for wti in wti_list if wti is not None])
         logger.info("Estimated %d translations out of %d images.", num_translations, num_images)
-        return wTi_list, ta_metrics
-
-
-def _sample_kde_directions(w_i2Ui1_measurements: BinaryMeasurementsUnit3, num_samples: int) -> List[Unit3]:
-    """Fits a Gaussian density kernel to the provided measurements, and then samples num_samples from this kernel.
-
-    Args:
-        w_i2Ui1_measurements: List of BinaryMeasurementUnit3 direction measurements.
-        num_samples: Number of samples to be sampled from the kernel.
-
-    Returns:
-        List of sampled Unit3 directions.
-    """
-    w_i2Ui1_list = [w_i2Ui1.measured() for w_i2Ui1 in w_i2Ui1_measurements]
-    if len(w_i2Ui1_list) > MAX_KDE_SAMPLES:
-        w_i2Ui1_subset_indices = np.random.choice(range(len(w_i2Ui1_list)), MAX_KDE_SAMPLES, replace=False).tolist()
-        w_i2Ui1_list = [w_i2Ui1_list[i] for i in w_i2Ui1_subset_indices]
-
-    w_i2Ui1_spherical = conversion_utils.cartesian_to_spherical_directions(w_i2Ui1_list)
-
-    # gaussian_kde expects each sample to be a column, hence transpose.
-    kde = stats.gaussian_kde(w_i2Ui1_spherical.T)
-    sampled_directions_spherical = kde.resample(size=num_samples).T
-    return conversion_utils.spherical_to_cartesian_directions(sampled_directions_spherical)
-
 
-def _sample_random_directions(num_samples: int) -> List[Unit3]:
-    """Samples num_samples Unit3 3D directions.
-    The sampling is done in 2D spherical coordinates (azimuth, elevation), and then converted to Cartesian coordinates.
-    Sampling in spherical coordinates was found to have precision-recall for 1dsfm outlier rejection.
-
-    Args:
-        num_samples: Number of samples required.
-
-    Returns:
-        List of sampled Unit3 directions.
-    """
-    sampled_azimuth = np.random.uniform(low=-np.pi, high=np.pi, size=(num_samples, 1))
-    sampled_elevation = np.random.uniform(low=0.0, high=np.pi, size=(num_samples, 1))
-    sampled_azimuth_elevation = np.concatenate((sampled_azimuth, sampled_elevation), axis=1)
-
-    return conversion_utils.spherical_to_cartesian_directions(sampled_azimuth_elevation)
+        # Combine estimated global rotations and global translations to Pose(3) objects.
+        wTi_list = [
+            Pose3(wRi, wti) if wRi is not None and wti is not None else None for wRi, wti in zip(wRi_list, wti_list)
+        ]
+        return wTi_list, ta_metrics
 
 
-def _get_measurement_angle_errors(
+def get_measurement_angle_errors(
     i1_i2_pairs: Set[Tuple[int, int]],
-    i2Ui1_measurements: Dict[Tuple[int, int], Unit3],
-    gt_i2Ui1_measurements: Dict[Tuple[int, int], Unit3],
-) -> List[Optional[float]]:
```

Comment:
why were there None's in here before, but not anymore?

Reply:
+1. There are be Nones in i2Ui1_measurements but not in gt_i2Ui1_measurements

Reply:
in theory if one of those measurements in None, the function can return a None. Do we want that?